utils/sparse.sh

```diff
@@ -35,12 +35,27 @@ RE="$RE|error: undefined identifier '__builtin_fpclassify'"
 search=$(gcc -print-search-dirs | awk '($1 == "install:"){print "-I" $2}')
 
 #
-# We're trying to use sparse against glibc headers which go wild trying to
-# use internal compiler macros to test features.  We copy gcc's and give
-# them to sparse.  But not __SIZE_TYPE__ 'cause sparse defines that one.
+# Filter out whatever gcc defines we already have from the sparse defines
+# so they don't throw redefined errors.
 #
 defines=".sparse.gcc-defines.h"
-gcc -dM -E -x c - < /dev/null | grep -v __SIZE_TYPE__ > $defines
+if [[ ! -f $defines ]]; then
```

Comment:
This now never recalculates the defines if the toolchain changes.  Get rid of the test and mv.  Append .$$ to the defines="" file name so each invocation uses its own file.  Then remove it after the sparse invocation.  It'll still have the prefix that'll be removed by make clean if it gets interrupted or crashes.